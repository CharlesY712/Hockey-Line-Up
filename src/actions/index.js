export const selectSeason = (season) => ({
  type: 'SELECT_SEASON',
  season
});