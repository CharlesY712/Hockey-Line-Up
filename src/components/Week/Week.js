import React from 'react';

const Week = () => {
  return (
    <div>Please select a week in the box above.</div>
  );
};

export default Week;