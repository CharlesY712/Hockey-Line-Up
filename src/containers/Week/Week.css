.date,
.directions {
  color: #677584;
}