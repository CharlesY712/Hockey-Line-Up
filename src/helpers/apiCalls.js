import { username, password } from '../apiKey';

const seasonName = '2016-2017-regular';
const base = 'https://api.mysportsfeeds.com/v1.2/pull/nhl';
const ext = `/${seasonName}/full_game_schedule.json`;
const url = base + ext;

const fetchSeason = async () => {
  const response = await fetch(url, {
    method: 'GET',
    headers: {
      "Authorization": "Basic " + btoa(username + ":" + password)
    }
  });
  const json = await response.json();
  return json;
};

export default fetchSeason;