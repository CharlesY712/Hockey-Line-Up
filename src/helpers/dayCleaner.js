const dayCleaner = (season, day) => {
  const games = season.reduce((gamesOnDay, game) => {
    if (game.date === day) {
      gamesOnDay = [game, ...game];
    }
    return gamesOnDay;
  }, []);
  return games;
};

export default dayCleaner;