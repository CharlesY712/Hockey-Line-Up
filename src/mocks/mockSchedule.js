export const schedule = {
  "dailygameschedule": {
    "lastUpdatedOn": "2018-04-12 4:24:49 PM",
    "gameentry": [
      {
        "id": "45866",
        "scheduleStatus": "Normal",
        "originalDate": null,
        "originalTime": null,
        "delayedOrPostponedReason": null,
        "date": "2018-04-14",
        "time": "3:00PM",
        "awayTeam": {
          "ID": "22",
          "City": "Colorado",
          "Name": "Avalanche",
          "Abbreviation": "COL"
        },
        "homeTeam": {
          "ID": "18",
          "City": "Nashville",
          "Name": "Predators",
          "Abbreviation": "NSH"
        },
        "location": "Bridgestone Arena"
      },
      {
        "id": "45875",
        "scheduleStatus": "Normal",
        "originalDate": null,
        "originalTime": null,
        "delayedOrPostponedReason": null,
        "date": "2018-04-14",
        "time": "3:00PM",
        "awayTeam": {
          "ID": "7",
          "City": "New Jersey",
          "Name": "Devils",
          "Abbreviation": "NJD"
        },
        "homeTeam": {
          "ID": "1",
          "City": "Tampa Bay",
          "Name": "Lightning",
          "Abbreviation": "TBL"
        },
        "location": "Amalie Arena"
      },
      {
        "id": "45874",
        "scheduleStatus": "Normal",
        "originalDate": null,
        "originalTime": null,
        "delayedOrPostponedReason": null,
        "date": "2018-04-14",
        "time": "8:00PM",
        "awayTeam": {
          "ID": "12",
          "City": "Toronto",
          "Name": "Maple Leafs",
          "Abbreviation": "TOR"
        },
        "homeTeam": {
          "ID": "11",
          "City": "Boston",
          "Name": "Bruins",
          "Abbreviation": "BOS"
        },
        "location": "TD Garden"
      },
      {
        "id": "45873",
        "scheduleStatus": "Normal",
        "originalDate": null,
        "originalTime": null,
        "delayedOrPostponedReason": null,
        "date": "2018-04-14",
        "time": "10:30PM",
        "awayTeam": {
          "ID": "26",
          "City": "San Jose",
          "Name": "Sharks",
          "Abbreviation": "SJS"
        },
        "homeTeam": {
          "ID": "29",
          "City": "Anaheim",
          "Name": "Ducks",
          "Abbreviation": "ANA"
        },
        "location": "Honda Center"
      }
    ]
  }
};