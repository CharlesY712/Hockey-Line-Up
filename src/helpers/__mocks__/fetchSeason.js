import { schedule } from '../../mocks/mockschedule';
import { emptySchedule } from '../../mocks/mockEmptySchedule';

export const fetchSeason = jest.fn()
  .mockImplementationOnce(() => emptySchedule )
  .mockImplementation(() => schedule );