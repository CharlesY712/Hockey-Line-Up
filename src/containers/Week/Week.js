import React, { Component } from 'react';
import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Game from '../../components/Game/Game';
import * as actions from '../../actions';
import { fetchSeason } from '../../helpers/fetchSeason';
import loadingGif from '../../images/icons/blue_loading.gif';
import './Week.css';

export class Week extends Component {
  constructor() {
    super();
    this.state = {
      games: [],
      isLoading: false
    };
  }

  componentDidMount() {
    if (this.props.date.includes('W')) {
      this.getWeekDays();
      this.setState({isLoading: true});
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps.date !== this.props.date) {
      if (this.props.date.includes('W')) {
        this.getWeekDays();
        this.setState({isLoading: true});
      }
    }
  }

  getDateOfISOWeek(year, week) {
    const simple = new Date(year, 0, 1 + (week - 1) * 7);
    const dow = simple.getDay();
    const ISOweekStart = simple;
    ISOweekStart.setDate(simple.getDate() + 8 - simple.getDay());
    return ISOweekStart.toISOString().slice(0, 10);
  }

  async getWeekDays() {
    const year = this.props.date.slice(0, 4);
    const week = this.props.date.slice(6, 8);
    const date = this.getDateOfISOWeek(year, week);
    let weekOfGames = [];
    for (let index = 0; index < 7; index++) {
      let myDate = new Date(date);
      myDate.setDate(myDate.getDate() + index);
      let nextDay = myDate.toISOString().slice(0, 10);
      const games = await fetchSeason(nextDay);
      const gameSchedule = games.dailygameschedule.gameentry;
      if (gameSchedule !== undefined) {
        weekOfGames = [...weekOfGames, ...gameSchedule];
      }
    }
    this.props.addSchedule(weekOfGames);
    const gameComponents = this.displayGames(weekOfGames);
    this.setState({games: gameComponents, isLoading: false});
  }

  displayGames() {
    const mappedGames = this.props.schedule.map(schedule => {
      return <Game
        key={schedule.id}
        homeTeamCity={schedule.homeTeam.City}
        homeTeamName={schedule.homeTeam.Name}
        awayTeamCity={schedule.awayTeam.City}
        awayTeamName={schedule.awayTeam.Name}
        date={schedule.date.slice(5)}
        time={schedule.time}
      />;
    });
    return mappedGames;
  }
  
  render() {
    if (this.props.date.includes('W')) {
      return (
        <div>
          {
            this.state.isLoading &&
          <section>
            <img className="loading-gif" src={loadingGif} alt="loading"/>
          </section>
          }
          {
            !this.state.isLoading &&
          <section>
            <div className="directions">Please select a week in the box above.</div>
            <h1 className="date">{this.props.date}</h1>
            {this.state.games}
          </section>
          }
        </div>
      );
    } else {
      return (
        <section>
          <div className="directions">Please select a week in the box above.</div>
        </section>
      );
    }
  }
}

Week.propTypes = {
  addSchedule: PropTypes.func,
  addScoreboard: PropTypes.func,
  date: PropTypes.string,
  setDate: PropTypes.func,
  schedule: PropTypes.array
};

export const mapStateToProps = state => ({
  date: state.setDate,
  schedule: state.schedule
});

export const mapDispatchToProps = dispatch => ({
  addSchedule: (season) => dispatch(actions.addSchedule(season)),
  setDate: (date) => dispatch(actions.setDate(date))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Week));
