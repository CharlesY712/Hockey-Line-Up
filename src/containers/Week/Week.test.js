import React from 'react';
import { shallow } from 'enzyme';
import { Week, mapStateToProps, mapDispatchToProps } from './Week';
import { schedule }  from '../../mocks/mockSchedule';
import { fetchSeason } from '../../helpers/fetchSeason';
import * as actions from '../../actions';

jest.mock('../../helpers/fetchSeason.js');
jest.mock('../../helpers/fetchScoreboard.js');

describe('Week', () => {

  describe('Week container', () => {
    const mockAddSchedule = jest.fn();
    const mockSetDate = jest.fn();
    let wrapper;

    it('should not display schedule on update if there are no games in Week', () => {
      wrapper = shallow(<Week
        date={'2018-W24'}
        schedule={schedule.dailygameschedule.gameentry}
        addSchedule={mockAddSchedule}
        setDate={mockSetDate}
      />);
      
      const mockPrevProps = {date: '2018-W24'};
      wrapper.instance().componentDidUpdate(mockPrevProps);

      expect(wrapper).toMatchSnapshot();
    });

    it('should not display games on mount if selected date doesnt include W', () => {
      wrapper = shallow(<Week
        date={'2018-04-12'}
        schedule={[schedule]}
        addSchedule={mockAddSchedule}
        setDate={mockSetDate}
      />);
      
      expect(wrapper).toMatchSnapshot();
    });

    it('should not display games on update if selected date doesnt inclued W', () => {
      wrapper = shallow(<Week
        date={'2018-04-12'}
        schedule={[schedule]}
        addSchedule={mockAddSchedule}
        setDate={mockSetDate}
      />);
      
      const mockPrevProps = {date: '2018-04-14'};
      wrapper.instance().componentDidUpdate(mockPrevProps);

      expect(wrapper).toMatchSnapshot();
    });

    it('should display schedule when a Week with games is selected', () => {
      wrapper = shallow(<Week
        date={'2018-W16'}
        schedule={schedule.dailygameschedule.gameentry}
        addSchedule={mockAddSchedule}
        setDate={mockSetDate}
      />);
      
      expect(fetchSeason).toHaveBeenCalled();
    });

    it('should display schedule when the component is updated', () => {
      wrapper = shallow(<Week
        date={'2018-W15'}
        schedule={schedule.dailygameschedule.gameentry}
        addSchedule={mockAddSchedule}
        setDate={mockSetDate}
      />);

      const mockPrevProps = {date: '2018-W14'};
      wrapper.instance().componentDidUpdate(mockPrevProps);

      expect(fetchSeason).toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    
    it('correctly maps the schedule to props', () => {
      const mockState = { schedule: [schedule] };
      const mapped = mapStateToProps(mockState);
      const expected = [schedule];

      expect(mapped.schedule).toEqual(expected);
    });

    it('correctly maps the date to props', () => {
      const date = '2018-04-12';
      const mockState = { setDate: date };
      const mapped = mapStateToProps(mockState);
      const expected = date;

      expect(mapped.date).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {

    it('should call dispatch with the correct params on addSchedule', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      const expected = actions.addSchedule(schedule);

      mapped.addSchedule(schedule);

      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });

    it('should call dispatch with the correct params on setDate', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      const expected = actions.setDate(schedule);

      mapped.setDate(schedule);

      expect(mockDispatch).toHaveBeenCalledWith(expected);
    });
  });
});