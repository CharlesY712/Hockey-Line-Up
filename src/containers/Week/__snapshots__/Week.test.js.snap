// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Week Week container should not display games on mount if selected date doesnt include W 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Week
    addSchedule={[Function]}
    date="2018-04-12"
    schedule={
        Array [
            Object {
              "dailygameschedule": Object {
                "gameentry": Array [
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "COL",
                      "City": "Colorado",
                      "ID": "22",
                      "Name": "Avalanche",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "NSH",
                      "City": "Nashville",
                      "ID": "18",
                      "Name": "Predators",
                    },
                    "id": "45866",
                    "location": "Bridgestone Arena",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "3:00PM",
                  },
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "NJD",
                      "City": "New Jersey",
                      "ID": "7",
                      "Name": "Devils",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "TBL",
                      "City": "Tampa Bay",
                      "ID": "1",
                      "Name": "Lightning",
                    },
                    "id": "45875",
                    "location": "Amalie Arena",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "3:00PM",
                  },
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "TOR",
                      "City": "Toronto",
                      "ID": "12",
                      "Name": "Maple Leafs",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "BOS",
                      "City": "Boston",
                      "ID": "11",
                      "Name": "Bruins",
                    },
                    "id": "45874",
                    "location": "TD Garden",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "8:00PM",
                  },
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "SJS",
                      "City": "San Jose",
                      "ID": "26",
                      "Name": "Sharks",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "ANA",
                      "City": "Anaheim",
                      "ID": "29",
                      "Name": "Ducks",
                    },
                    "id": "45873",
                    "location": "Honda Center",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "10:30PM",
                  },
                ],
                "lastUpdatedOn": "2018-04-12 4:24:49 PM",
              },
            },
          ]
    }
    setDate={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="directions"
>
        Please select a week in the box above.
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Please select a week in the box above.",
        "className": "directions",
      },
      "ref": null,
      "rendered": "Please select a week in the box above.",
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="directions"
>
          Please select a week in the box above.
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please select a week in the box above.",
          "className": "directions",
        },
        "ref": null,
        "rendered": "Please select a week in the box above.",
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Week Week container should not display games on update if selected date doesnt inclued W 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Week
    addSchedule={[Function]}
    date="2018-04-12"
    schedule={
        Array [
            Object {
              "dailygameschedule": Object {
                "gameentry": Array [
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "COL",
                      "City": "Colorado",
                      "ID": "22",
                      "Name": "Avalanche",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "NSH",
                      "City": "Nashville",
                      "ID": "18",
                      "Name": "Predators",
                    },
                    "id": "45866",
                    "location": "Bridgestone Arena",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "3:00PM",
                  },
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "NJD",
                      "City": "New Jersey",
                      "ID": "7",
                      "Name": "Devils",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "TBL",
                      "City": "Tampa Bay",
                      "ID": "1",
                      "Name": "Lightning",
                    },
                    "id": "45875",
                    "location": "Amalie Arena",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "3:00PM",
                  },
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "TOR",
                      "City": "Toronto",
                      "ID": "12",
                      "Name": "Maple Leafs",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "BOS",
                      "City": "Boston",
                      "ID": "11",
                      "Name": "Bruins",
                    },
                    "id": "45874",
                    "location": "TD Garden",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "8:00PM",
                  },
                  Object {
                    "awayTeam": Object {
                      "Abbreviation": "SJS",
                      "City": "San Jose",
                      "ID": "26",
                      "Name": "Sharks",
                    },
                    "date": "2018-04-14",
                    "delayedOrPostponedReason": null,
                    "homeTeam": Object {
                      "Abbreviation": "ANA",
                      "City": "Anaheim",
                      "ID": "29",
                      "Name": "Ducks",
                    },
                    "id": "45873",
                    "location": "Honda Center",
                    "originalDate": null,
                    "originalTime": null,
                    "scheduleStatus": "Normal",
                    "time": "10:30PM",
                  },
                ],
                "lastUpdatedOn": "2018-04-12 4:24:49 PM",
              },
            },
          ]
    }
    setDate={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="directions"
>
        Please select a week in the box above.
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Please select a week in the box above.",
        "className": "directions",
      },
      "ref": null,
      "rendered": "Please select a week in the box above.",
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="directions"
>
          Please select a week in the box above.
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please select a week in the box above.",
          "className": "directions",
        },
        "ref": null,
        "rendered": "Please select a week in the box above.",
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Week Week container should not display schedule on update if there are no games in Week 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Week
    addSchedule={[Function]}
    date="2018-W24"
    schedule={
        Array [
            Object {
              "awayTeam": Object {
                "Abbreviation": "COL",
                "City": "Colorado",
                "ID": "22",
                "Name": "Avalanche",
              },
              "date": "2018-04-14",
              "delayedOrPostponedReason": null,
              "homeTeam": Object {
                "Abbreviation": "NSH",
                "City": "Nashville",
                "ID": "18",
                "Name": "Predators",
              },
              "id": "45866",
              "location": "Bridgestone Arena",
              "originalDate": null,
              "originalTime": null,
              "scheduleStatus": "Normal",
              "time": "3:00PM",
            },
            Object {
              "awayTeam": Object {
                "Abbreviation": "NJD",
                "City": "New Jersey",
                "ID": "7",
                "Name": "Devils",
              },
              "date": "2018-04-14",
              "delayedOrPostponedReason": null,
              "homeTeam": Object {
                "Abbreviation": "TBL",
                "City": "Tampa Bay",
                "ID": "1",
                "Name": "Lightning",
              },
              "id": "45875",
              "location": "Amalie Arena",
              "originalDate": null,
              "originalTime": null,
              "scheduleStatus": "Normal",
              "time": "3:00PM",
            },
            Object {
              "awayTeam": Object {
                "Abbreviation": "TOR",
                "City": "Toronto",
                "ID": "12",
                "Name": "Maple Leafs",
              },
              "date": "2018-04-14",
              "delayedOrPostponedReason": null,
              "homeTeam": Object {
                "Abbreviation": "BOS",
                "City": "Boston",
                "ID": "11",
                "Name": "Bruins",
              },
              "id": "45874",
              "location": "TD Garden",
              "originalDate": null,
              "originalTime": null,
              "scheduleStatus": "Normal",
              "time": "8:00PM",
            },
            Object {
              "awayTeam": Object {
                "Abbreviation": "SJS",
                "City": "San Jose",
                "ID": "26",
                "Name": "Sharks",
              },
              "date": "2018-04-14",
              "delayedOrPostponedReason": null,
              "homeTeam": Object {
                "Abbreviation": "ANA",
                "City": "Anaheim",
                "ID": "29",
                "Name": "Ducks",
              },
              "id": "45873",
              "location": "Honda Center",
              "originalDate": null,
              "originalTime": null,
              "scheduleStatus": "Normal",
              "time": "10:30PM",
            },
          ]
    }
    setDate={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section>
          <img
                    alt="loading"
                    className="loading-gif"
                    src="blue_loading.gif"
          />
</section>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="loading"
            className="loading-gif"
            src="blue_loading.gif"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "loading",
            "className": "loading-gif",
            "src": "blue_loading.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "section",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section>
            <img
                        alt="loading"
                        className="loading-gif"
                        src="blue_loading.gif"
            />
</section>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="loading"
              className="loading-gif"
              src="blue_loading.gif"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "loading",
              "className": "loading-gif",
              "src": "blue_loading.gif",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "section",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
